pkg/rc/replication_controller.go

```diff
@@ -295,7 +293,11 @@ func (rc *replicationController) addPods(current types.PodLocations, eligible []
 				return util.Errorf("could not schedule pods due to transaction violation: %s", transaction.TxnErrorsToString(resp.Errors))
 			}
 
-			return util.Errorf(errMsg)
+			if err != nil {
+				return util.Errorf("Could not meet desires by allocating nodes: %s. Scheduled on %d nodes instead.", err, i)
+			}
+
+			return nil
```

Comment:
I think you don't want to return here because you haven't done the `schedule()` yet